### PR TITLE
Instrument scrape pool `sync()`

### PR DESCRIPTION
**Pull request by stuartnelson3**
Instruments:
- duration
- count

@fabxc

### CODE REVIEW

File: retrieval/scrape.go [hunk 1]
```diff
@@ -74,12 +74,29 @@ var (
 		},
 		[]string{interval},
 	)
+	targetSyncIntervalLength = prometheus.NewSummary(
```
Comment (fabxc) [inline retrieval/scrape.go]: There's one scrape pool per job. Having the a `scrape_job` dimension with that would be helpful.

Comment (stuartnelson3) [inline retrieval/scrape.go]: Sounds good. Would there be a benefit of using a histogram instead of a summary?

Comment (brian-brazil) [inline retrieval/scrape.go]: Given how often it happens (likely no more than once per scrape interval), a histogram wouldn't gain that much.

Comment (fabxc) [inline retrieval/scrape.go]: Jobs are few enough that aggregating over them doesn't add much. Detecting slow ones here by eye is sufficient.

File: retrieval/scrape.go [hunk 2]
```diff
@@ -74,12 +75,30 @@ var (
...
+	targetSyncIntervalLength = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace:  namespace,
+			Name:       "target_sync_length_seconds",
+			Help:       "Actual interval to sync the scrape pool.",
+			Objectives: map[float64]float64{0.01: 0.001, 0.05: 0.005, 0.5: 0.05, 0.90: 0.01, 0.99: 0.001},
+		},
+		[]string{scrapeJob},
+	)
+	targetScrapePoolSyncsCounter = prometheus.NewCounter(
```
Comment (fabxc) [inline retrieval/scrape.go]: Symmetrically this should also have the `scrape_job` label.

**Comment by fabxc** [comment]
Thanks :+1:

**Pull request merged**